### ISSUE TITLE
 If deep learning is the answer, what is the question? 

### ISSUE THREAD

**Issue by xuanyuw** [xuanyuw/xuanyuw.github.io#33]
# If deep learning is the answer, what is the question?

{
last author: [[Christopher Summerfield]]
keyword: [RNN, Perspectives]
parent: [[Topic-Machine  Learning]]
paper: [https://doi.org/10.1038/s41593-021-00840-6](https://doi.org/10.1038/s41593-021-00840-6)
}
#perspectives

## New concepts

-**Connectionism**: an approach in cognitive science that hopes to explain mental phenomena using artificial neural networks.

## Main points

- "Deep Learning is the answer":
  - Network w/ biologically plausible constraints
    - Sophisticated behavior and biologically plausible neural representations can emerge from a limited set of computational principles of an appropriately optimized and well-trained network. (CNN v.s. V1-V3 feature extraction)
  - Network w/o constraints
    - underlying idea: both unconstrained NN and the brain develop functional adaptations without meaningful constraints. Neural systems are fundamentally uninterpretable.
    - Population-level activity patterns (e.g., got from dimensionality reduction) meaningfully distinguish experimental variables.
    - Argues that computational model is explainable at the aggregated level (driven by network structures and hyperparameters), explaining single-neuron-level computation principles is futile.
- Generating hypothesis
  - How to understand the mapping from design to performance in deep networks?
- Can deep networks be neural models?
  - Overstated equivalence
    - Popular approach:
      - learn a linear mapping from network units to neurons, and evaluate the predictive validity of the resulting regression model in a held-out dataset. But the performance is not really equivalent
        - a linear relationship can also exist even if the sparsity and dimensionality are different
          - Need tests that restrict the freedom of the mapping function: representational similarity analysis.
    - Problem:
      - Equivalent representations in deep networks and the brain are based on deep networks that were not trained with GD.
    - Future goal:
      - assess if deep networks and biological brains learn in similar ways (over learning process)
      - test the equivalence of patterns of response between biological and artificial systems
        - human and machines make different mistakes. And human generalizes better
- Learning rules for perception
  - whether neural codes could emerge from a learning principle applied to a relatively generic architecture and starting point
  - unsupervised learning principle v.s. Supervised learning principles to predict more accurately.
  - Evaluate learning rules:
    - test the model on naturalistic data
      - need a lot of labeled data
    - find a principle that generates bio-plausible representational changes during learning. (more direct)
    - both methods might rely on BP with GD, but a biologically realistic implementation is needed for GD.
- Deep learning for cognition
  - Human and animals: - concept abstraction, rapid knowledge acquisition, and relational knowledge structures
    Combining deep learning with other methods (reinforcement learning, content-addressable memory, or Monte Carlo tree search) can be a POC for end-to-end learning in complex cognitive architectures.
  - One approach: focus on the tasks that humans and the network perform differently. See how the human solve the problem and see how the learning rule or architecture may be varied to accommodate the same behavior
- Abstraction and generalization
  - Main question:
    - How biological brains encode, compose and generalize abstract knowledge
  - Obstacles:
    - It is unknown whether the experimental animals have the neural mechanisms for knowledge characteristics like human
    - Humans and other animals have diverse prior (different exposure and potentially different learning methods) before the experiment
    - Humans and machines learn over very different timescales
  - One possibility:
    - lifelong exposure to sensory data might allow strong invariances to emerge via (supervised or unsupervised) learning
- Resource allocation in learning
  - Deep networks have "catastrophic forgetting," but humans don't.
    - Possible neural mechanisms:
      - (Hippocampus) episodic memories --> replay --> consolidates into neocortical circuits
      - some deep networks have analogous methods
    - Machine learning solutions
      - identifying a weight subspace in which new learning is least likely to cause retrospective inference ('freezing' certain synapses)
      - unsupervised processes facilitate continual learning in biological systems by clustering neural representations according to their context.
    - More general question on resource allocation:
      - How do biological systems evolve both to minimize interference and maximize generalization?
        - Capacity limits in the biological control process
  - Understanding deep networks
    - Main advantage of deep networks: can access every synaptic weight and unit activation over the course of learning.
      - Better to look at the learning dynamics instead of the final stage.
    - Idealized network model can tell us something about learning (e.g., semantic cognition)
    - The deep network model must be interpretable while explaining biological phenomena.
  
## My Random Questions and Ideas

- About the multimodule RNN project
  1. How to combine DNN with reinforcement learning (and others) in the project (or hebbian learning rule)?
  2. Possible to make the network unsupervised?
  3. "Freezing synapses" to make leanring faster and more stable?
  4. Bio-plausible implementation of gradient descent?
  5. Increase network size to improve stability?
  6. Looking at the learning process of decision making? (compare with traininfg data if possible?)
  7. Is there another way to simplify the network?
  8. Is the network over-parameterized?
- Objectives
  - How to address the overstated equivalence issues? I.e., if I'm using RNN to mimic this system, how should I convince others that this system is biologically plausible enough?
  - What neural principles will the end goal of this project reveal?
